src/GafferScene/Preview/RendererAlgo.cpp

```diff
@@ -157,9 +157,8 @@ class LocationTask : public tbb::task
 template <class Functor>
 void parallelProcessLocations( const GafferScene::ScenePlug *scene, Functor &f )
 {
-	Gaffer::ContextPtr c = new Gaffer::Context( *Gaffer::Context::current(), Gaffer::Context::Borrowed );
-	GafferScene::Filter::setInputScene( c.get(), scene );
-	LocationTask<Functor> *task = new( tbb::task::allocate_root() ) LocationTask<Functor>( scene, c.get(), ScenePlug::ScenePath(), f );
+	FilterPlug::SceneScope sceneScope( Gaffer::Context::current(), scene );
+	LocationTask<Functor> *task = new( tbb::task::allocate_root() ) LocationTask<Functor>( scene, Gaffer::Context::current(), ScenePlug::ScenePath(), f );
```

Comment:
I haven't done this one. One of the future optimisations I'm considering is that in some cases EditableScope might not even copy-construct the original context, but instead it would make in-place changes and then undo them at destruction. I felt like a `context()` method implies that the Scope owns a context of its own, so I'd like to give things some time to see how they pan out if it's OK...